File: app/typescript/src/domain/model/audit/AuditAction.ts
// src/audit/domain/AuditAction.ts

/**
 * 監査ログのアクション種別
 */
export enum AuditAction {
  CREATE = 'CREATE', // 作成
  UPDATE = 'UPDATE', // 更新
  DELETE = 'DELETE', // 削除
  READ = 'READ' // 参照（必要に応じて）
}

File: app/csharp/AccountingSystem.Domain/Models/Audit/AuditAction.cs
namespace AccountingSystem.Domain.Models.Audit;

/// <summary>
/// 監査ログのアクション種別
/// </summary>
public enum AuditAction
{
    CREATE,
    UPDATE,
    DELETE
}

public static class AuditActionExtensions
{
    public static string GetDisplayName(this AuditAction action)
    {
        return action switch
        {
            AuditAction.CREATE => "作成",
            AuditAction.UPDATE => "更新",
            AuditAction.DELETE => "削除",
            _ => throw new ArgumentException($"Unknown action: {action}")
        };
    }
}
